test/mocks/auth/firebase_auth_mocks.dart

```diff
@@ -13,13 +15,24 @@ class FakeFirebaseAuthPeriodic extends Fake implements FirebaseAuth {
       });
 }
 
-class FakeFirebaseAuthWithError extends Fake implements FirebaseAuth {
+class FakeFirebaseAuthSignInException extends Fake implements FirebaseAuth {
   @override
-  Stream<auth.User> authStateChanges() => Stream.fromIterable([FakeAuthUser()]);
+  Future<UserCredential> signInWithCredential(AuthCredential credential) =>
+      throw FirebaseAuthException(
+          code: 'user-not-found', message: 'test error: cant find user');
+}
 
+class FakeFirebaseAuthThrows extends Fake implements FirebaseAuth {
   @override
   Future<UserCredential> signInWithCredential(AuthCredential credential) =>
-      null;
+      throw Exception('firebaseAuth.signIn');
+  @override
+  Future<void> signOut() => throw Exception('firebaseAuth.signOut');
+
+  @override
+  Future<auth.UserCredential> signInWithEmailAndPassword(
+          {String email, String password}) =>
+      throw Exception('firebaseAuth.signInWithEmailAndPassword');
 }
 
 class FakeFirebaseAuth1 extends Fake implements FirebaseAuth {
```

Comment:
I know this wasn't commented before (that's my bad) but would you mind adding a brief comment that indicates what the class is for and how/where it's used?